launcher/modplatform/flame/FlamePackExportTask.cpp

```diff
@@ -0,0 +1,440 @@
+    setProgress(0, 0);
+    collectFiles();
+}
+
+bool FlamePackExportTask::abort()
+{
+    if (task != nullptr) {
+        task->abort();
+        task = nullptr;
+        emitAborted();
+        return true;
+    }
+
+    if (buildZipFuture.isRunning()) {
+        buildZipFuture.cancel();
+        // NOTE: Here we don't do `emitAborted()` because it will be done when `buildZipFuture` actually cancels, which may not occur
+        // immediately.
+        return true;
+    }
+
+    return false;
+}
+
+void FlamePackExportTask::collectFiles()
+{
+    setAbortable(false);
+    QCoreApplication::processEvents();
+
+    files.clear();
+    if (!MMCZip::collectFileListRecursively(instance->gameRoot(), nullptr, &files, filter)) {
+        emitFailed(tr("Could not search for files"));
+        return;
+    }
+
+    pendingHashes.clear();
+    resolvedFiles.clear();
+
+    if (mcInstance != nullptr) {
+        mcInstance->loaderModList()->update();
+        connect(mcInstance->loaderModList().get(), &ModFolderModel::updateFinished, this, &FlamePackExportTask::collectHashes);
+    } else
+        collectHashes();
+}
+
+void FlamePackExportTask::collectHashes()
+{
+    setAbortable(true);
+    setStatus(tr("Find file hashes..."));
+    auto mods = mcInstance->loaderModList()->allMods();
+    ConcurrentTask::Ptr hashing_task(new ConcurrentTask(this, "MakeHashesTask", 10));
+    task.reset(hashing_task);
+    int totalProgres = mods.count();
+    setProgress(0, totalProgres);
+    for (auto* mod : mods) {
+        if (!mod || mod->type() == ResourceType::FOLDER) {
+            setProgress(m_progress + 1, totalProgres);
+            continue;
+        }
+        if (mod->metadata() && mod->metadata()->provider == ModPlatform::ResourceProvider::FLAME) {
+            resolvedFiles.insert(mod->fileinfo().absoluteFilePath(),
+                                 { mod->metadata()->project_id.toInt(), mod->metadata()->file_id.toInt(), mod->enabled(), true,
+                                   mod->metadata()->name, mod->metadata()->slug, mod->authors().join(", ") });
+            setProgress(m_progress + 1, totalProgres);
+            continue;
+        }
+
+        auto hash_task = Hashing::createFlameHasher(mod->fileinfo().absoluteFilePath());
+        connect(hash_task.get(), &Hashing::Hasher::resultsReady, [this, mod, totalProgres](QString hash) {
+            if (m_state == Task::State::Running) {
+                setProgress(m_progress + 1, totalProgres);
+                pendingHashes.insert(hash, { mod->name(), mod->fileinfo().absoluteFilePath(), mod->enabled(), true });
+            }
+        });
+        connect(hash_task.get(), &Task::failed, this, &FlamePackExportTask::emitFailed);
+        hashing_task->addTask(hash_task);
+    }
+
+    for (const QFileInfo& file : files) {
+        const QString relative = gameRoot.relativeFilePath(file.absoluteFilePath());
+        if (!relative.endsWith(".zip") || !relative.startsWith("resourcepacks/"))
```

Comment:
:P maybe you could try to mimic mrpack more

Reply:
this should be quite similar to mrpack: here I'm interested only in resourcepacks so I look only at that folder and that they have a specific extension.
The mrpack does the same but with multiple resource types here:(https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/modplatform/modrinth/ModrinthPackExportTask.cpp#L97-L108)
As I know CurseForge only has mods and resource packs in a mod pack(correct me if I'm wrong), so I chose to parse the mods first(using all the mods list) and then parse the resource packs separately by parsing selected files

Reply:
in the end did make it more similar

Reply:
> this should be quite similar to mrpack: here I'm interested only in resourcepacks so I look only at that folder and that they have a specific extension. The mrpack does the same but with multiple resource types here:(https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/modplatform/modrinth/ModrinthPackExportTask.cpp#L97-L108) As I know CurseForge only has mods and resource packs in a mod pack(correct me if I'm wrong), so I chose to parse the mods first(using all the mods list) and then parse the resource packs separately by parsing selected files

The problem is it's checking for things which are either .zip or in resourcepacks

Reply:
zips can also be mods

Reply:
there's no condition for jar

Reply:
I feel like your previous code was better than in mrpack now, however, it should be && rather than ||

Reply:
sorry I did forget old code there now it should be ok